jaxrs-clients/src/test/java/com/palantir/remoting3/jaxrs/JaxRsClientFailoverTest.java

```diff
@@ -79,17 +79,26 @@ public void testConsecutiveCalls() throws Exception {
     }
 
     @Test
-    public void testFailoverOnDnsFailure() throws Exception {
+    public void testConnectionError_performsFailoverOnDnsFailure() throws Exception {
         server1.enqueue(new MockResponse().setBody("\"foo\""));
 
-        TestService proxy = JaxRsClient.create(TestService.class, "agent",
+        TestService bogusHostProxy = JaxRsClient.create(TestService.class, "agent",
                 ClientConfiguration.builder()
                         .from(createTestConfig(
                                 "http://foo-bar-bogus-host.unresolvable:80",
                                 "http://localhost:" + server1.getPort()))
                         .maxNumRetries(1)
                         .build());
-        assertThat(proxy.string(), is("foo"));
+        assertThat(bogusHostProxy.string(), is("foo"));
         assertThat(server1.getRequestCount(), is(1));
     }
+
+    @Test
+    public void testQosError_performsFailover() throws Exception {
+        server1.enqueue(new MockResponse().setResponseCode(503));
+        server1.enqueue(new MockResponse().setBody("\"foo\""));
```

Comment:
Do we not guarantee not retrying the same server twice?

Reply:
No.